core/output/interactive_test.go

```diff
@@ -252,4 +253,24 @@ func TestInteractiveShellOutput(t *testing.T) {
 			assert.Equal(expectedFinalOutputLines[i], finalLine)
 		}
 	})
+
+	t.Run("concurrent", func(t *testing.T) {
+		output := output.NewInteractiveShell(io.Discard, io.Discard, 0)
+
+		wg := sync.WaitGroup{}
+		doStuff := func() {
+			output.StartOperation("working")
+			output.Info("a message")
+			output.EndOperation(true)
+			output.StartOperation("working")
+			output.Error(nil, "an error")
+			output.EndOperation(false)
+			wg.Done()
+		}
+
+		wg.Add(2)
+		go doStuff()
+		go doStuff()
+		wg.Wait()
+	})
```

Comment:
There don't seem to be any assertions in here - what order are you expecting here? I'm not sure what this test proves.

Reply:
I think that go race detector would complain here if the code wasn't patched with a mutex.

Reply:
Yes, exactly.